### PR TITLE
feat: create component to load screen.

### PR DESCRIPTION
**Pull request by uziasferreirazup**
### Related Issues

#1309

### Description and Example

Remove the beagle remove view and create a new widget called BeagleWidget will be used to render some view in the beagle.

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/03b570620268a4cb4925e4d3a2333e35a93a7de0/android/sample/src/main/java/br/com/zup/beagle/sample/widgets/Text.kt#L19">android/sample/src/main/java/br/com/zup/beagle/sample/widgets/Text.kt#L19</a> - The import 'android.graphics.Color' is unused.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/03b570620268a4cb4925e4d3a2333e35a93a7de0/android/sample/src/main/java/br/com/zup/beagle/sample/widgets/Text.kt#L22">android/sample/src/main/java/br/com/zup/beagle/sample/widgets/Text.kt#L22</a> - The import 'br.com.zup.beagle.android.utils.observeBindChanges' is unused.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          5 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/1723d484f927f5699612e3847b804e6e8bd42276/beagle/lib/beagle_stream_builder.dart#L29">beagle/lib/beagle_stream_builder.dart#L29</a> - Name non-constant identifiers using lowerCamelCase</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/1723d484f927f5699612e3847b804e6e8bd42276/beagle/lib/beagle_stream_builder.dart#L31">beagle/lib/beagle_stream_builder.dart#L31</a> - Name non-constant identifiers using lowerCamelCase</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/1723d484f927f5699612e3847b804e6e8bd42276/beagle/lib/beagle_stream_builder.dart#L33">beagle/lib/beagle_stream_builder.dart#L33</a> - Name non-constant identifiers using lowerCamelCase</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/1723d484f927f5699612e3847b804e6e8bd42276/beagle/lib/beagle_stream_builder.dart#L36">beagle/lib/beagle_stream_builder.dart#L36</a> - Name non-constant identifiers using lowerCamelCase</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/1723d484f927f5699612e3847b804e6e8bd42276/beagle/lib/beagle_stream_builder.dart#L38">beagle/lib/beagle_stream_builder.dart#L38</a> - Name non-constant identifiers using lowerCamelCase</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_flutter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/ServerDrivenActivity.kt [hunk 1]
```diff
@@ -20,6 +20,7 @@ import android.os.Bundle
 import android.view.View
 import androidx.appcompat.widget.Toolbar
 import br.com.zup.beagle.R
+import br.com.zup.beagle.android.annotation.RegisterController
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/ServerDrivenActivity.kt:23]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9977873c9da8f7dc9e3674a94d24cd2c58111d61/android/beagle/src/main/java/br/com/zup/beagle/android/view/ServerDrivenActivity.kt#L23"></span>The import 'br.com.zup.beagle.android.annotation.RegisterController' is unused.</td>
    </tr>
  </tbody>
</table>

File: flutter/beagle/lib/components/beagle_undefined_widget.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        super(key: key);
+
+  /// [BeagleEnvironment] that will provide current environment.
+  final BeagleEnvironment _environment;
+
+  @override
+  Widget build(BuildContext context) {
+    return _environment == BeagleEnvironment.debug
+        ? const Text('Undefined Component')
+        : const SizedBox(height: 0.01);
```
Comment (matheusribeirozup) [inline flutter/beagle/lib/components/beagle_undefined_widget.dart:35]: Why not an empty Container?

File: flutter/beagle/test/components/beagle_undefined_widget_test.dart [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      testWidgets('Then it should have the correct text',
+          (WidgetTester tester) async {
+        await tester.pumpWidget(createWidget());
+
+        final textFinder = find.text(text);
+
+        expect(textFinder, findsOneWidget);
+      });
+
+      testWidgets('Then it should not have text widget',
```
Comment (matheusribeirozup) [inline flutter/beagle/test/components/beagle_undefined_widget_test.dart:46]: What do you think to put this in another group to better describe the test?
```suggestion
    group('When the environment is production', () {
        testWidgets('Then it should not have text widget',
```

File: flutter/beagle/lib/components/beagle_undefined_widget.dart (hunk 2 above)
Comment (uziasferreirazup) [reply flutter/beagle/lib/components/beagle_undefined_widget.dart:35]: an empty container has a large size, in which case it is not to take any space on the screen

File: flutter/beagle/test/components/beagle_undefined_widget_test.dart (hunk 3 above)
Comment (uziasferreirazup) [reply flutter/beagle/test/components/beagle_undefined_widget_test.dart:48]: done

**Review by matheusribeirozup — APPROVED** [review]

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | Classname | Time|
--- | --- | ---|
Should call Analytics Service with correct parameters | br.com.zup.beagle.android.view.viewmodel.AnalyticsViewModelTest$CreateActionReport | 0.259|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Should call Analytics Service with correct parameters | br.com.zup.beagle.android.view.viewmodel.AnalyticsViewModelTest$CreateScreenReport | 0.256|


<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by paulomeurerzup — APPROVED** [review]

**Pull request merged**